fe/fe-core/src/main/java/com/starrocks/analysis/Expr.java

```diff
@@ -286,6 +289,32 @@ public Type getOriginType() {
         return originType;
     }
 
+    private Optional<Expr> replaceLargeStringLiteralImpl() {
+        if (this instanceof LargeStringLiteral) {
+            return Optional.of(new StringLiteral(((LargeStringLiteral) this).getValue()));
+        }
+        if (children == null || children.isEmpty()) {
+            return Optional.empty();
+        }
+        List<Pair<Expr, Optional<Expr>>> childAndNewChildPairList = children.stream()
+                .map(child -> Pair.create(child, child.replaceLargeStringLiteralImpl()))
+                .collect(Collectors.toList());
+        if (childAndNewChildPairList.stream().noneMatch(p -> p.second.isPresent())) {
+            return Optional.empty();
+        }
+
+        for (int i = 0; i < this.children.size(); ++i) {
+            Pair<Expr, Optional<Expr>> childPair = childAndNewChildPairList.get(i);
+            Expr newChild = childPair.second.orElse(childPair.first);
+            setChild(i, newChild);
+        }
+        return Optional.of(this);
+    }
+
+    public Expr replaceLargeStringLiteral() {
```

Comment:
so `Expr` to SQL also needs to be fixed like this(eg:`AstToStringBuilder` )?

MV's defined sql is deduced from `AstToSQLBuilder#toSQL` which can also be wrong if it meets `LargeStringLiteral`?

Reply:
yeah, it seems refresh sql and mv sql also have this issue.